pennylane_sf/expectations.py

```diff
@@ -206,7 +211,7 @@ def poly_xp(state, wires, params):
 
     # HACK, we need access to the Poly instance in order to expand the matrix!
     op = pennylane.ops.PolyXP(Q, wires=wires, do_queue=False)
-    Q = op.heisenberg_obs(state.num_modes)
+    Q = op.heisenberg_obs(all_wires)
```

Comment:
Saw the [comment related to why all wires need to be passed to `heisenberg_obs`](https://github.com/PennyLaneAI/pennylane/pull/666/files#r458117642), very thorough thinking there! :blush:

Still trying to put together the details, one thing that came to mind is, could the QNode extract the `wire_indices` and then `heisenberg_obs` would "blindly" operate using those indices? One advantage would be that there's less complexity for `heisenberg_obs` and an object that is mostly the caller to it (QNode) could take this burden. Even if it would work, perhaps not too pressing as such a shift could be nice with several use cases not just `heisenberg_obs` :thinking:

Reply:
I'm not sure what is the best way to deal with ``operator`` functions that need to know how wires are arranged on a register. You are right, for ``heisenberg_obs`` we could reduce the number of arguments passed around by using wire indices directly. But in the long run I would almost prefer moving all those functions into the qnode or better, the device, since they are part of applying operators to circuits.

Reply:
Actually, @josh146 what do you think about this?

Reply:
I have to admit, I am not that up to date with the discussion, but my understanding is that the ordering comes into play due to the `heisenberg_expand` method, which takes the subset of wires and the _total_ number of wires, and tries to expand the operator to act on the full system (by inserting identities where required).

It definitely feels like logic that should be a part of the `CVQNode`.